src/MICore/Checksum.cs

```diff
@@ -0,0 +1,123 @@
+﻿using System;
+using System.Collections;
+using System.Collections.Generic;
+using System.Globalization;
+using System.IO;
+using System.Linq;
+using System.Security.Cryptography;
+using System.Text;
+using System.Threading.Tasks;
+
+namespace MICore
+{
+    public class Checksum
+    {
+#if CORECLR
+        private string _checksumString = null;
+        public byte[] Bytes { get; private set; } = null;
```

Comment:
Not that this is the end of the world, but does this need to be public? Always nice to keep mutable collections private if one can...